### ISSUE TITLE
Error when creating a SkyCoord object.

### ISSUE THREAD

**Issue by EricDepagne** [astropy/astropy#3399]
The combination of an astroquery.Simbad.simbad() and the creation of a SkyCoord object yields some strange results depending on the use of python3 or python 2.

The following works fine with (i)python2 :

```
In [3]: star2 = 'BPS CS 22878-027'
In [4]: rid2 = Simbad.query_object(star2)
In [5]: ra2 = rid2['RA']
In [6]: dec2 = rid2['DEC']
In [7]: c2 = SkyCoord(ra2, dec2, frame='icrs', unit=(u.hourangle, u.deg))
In [8]: star ='HE 1506-0113'
In [9]: rid = Simbad.query_object(star)
In [10]: ra = rid['RA']
In [11]: dec = rid['DEC']
In [12]: c = SkyCoord(ra, dec, frame='icrs', unit=(u.hourangle, u.deg))
```

whereas with (i)python 3, the following works:

```
In [10]: star2 = 'BPS CS 22878-027'
In [11]: rid2 = Simbad.query_object(star2)
In [12]: ra2 = rid2['RA']
In [13]: dec2 = rid2['DEC']
In [14]: c2 = SkyCoord(ra2, dec2, frame='icrs', unit=(u.hourangle, u.deg))
```

But the other one :
```
In [5]: star ='HE 1506-0113'
In [6]: rid = Simbad.query_object(star)
In [7]: ra = rid['RA']
In [8]: dec = rid['DEC']
In [9]: c = SkyCoord(ra, dec, frame='icrs', unit=(u.hourangle, u.deg))
```

produces the following error:

https://gist.github.com/EricDepagne/bc3b2224301c4dd46471
```
---------------------------------------------------------------------------
ValueError Traceback (most recent call last)
<ipython-input-9-8e80904223ec> in <module>()
----> 1 c = SkyCoord(ra, dec, frame='icrs', unit=(u.hourangle, u.deg))
 
/usr/local/lib/python3.4/dist-packages/astropy/coordinates/sky_coordinate.py in __init__(self, *args, **kwargs)
148
149 # Finally make the internal coordinate object.
--> 150 self._sky_coord_frame = frame.__class__(**coord_kwargs)
151
152 if not self._sky_coord_frame.has_data:
 
/usr/local/lib/python3.4/dist-packages/astropy/coordinates/baseframe.py in __init__(self, *args, **kwargs)
445 if (self.representation == SphericalRepresentation and
446 'distance' not in repr_kwargs):
--> 447 representation_data = UnitSphericalRepresentation(**repr_kwargs)
448 else:
449 representation_data = self.representation(**repr_kwargs)
 
/usr/local/lib/python3.4/dist-packages/astropy/coordinates/representation.py in __init__(self, lon, lat, copy)
425 lon, lat = broadcast_quantity(lon, lat)
426 except ValueError:
--> 427 raise ValueError("Input parameters lon and lat cannot be broadcast")
428
429 self._lon = lon
 
ValueError: Input parameters lon and lat cannot be broadcast
```

**Comment by mboquien** [comment]
To give more detail the ra and dec returned by astroquery are byte strings, which SkyCoords does not like at all. A minimal testcase could be:

In [4]: SkyCoord('15 09 14.3', '-01 24 57', frame='icrs', unit=(u.hourangle, u.deg))
Out[4]: <SkyCoord (ICRS): ra=227.3095833333333 deg, dec=-1.4158333333333333 deg>

In [5]: SkyCoord(b'15 09 14.3', b'-01 24 57', frame='icrs', unit=(u.hourangle, u.deg))
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)                                                                               
<ipython-input-5-17ea7e05626a> in <module>()                                                                                                                                                                                                
----> 1 SkyCoord(b'15 09 14.3', b'-01 24 57', frame='icrs', unit=(u.hourangle, u.deg))                                                                                                                                                      
                                                                                                                                                                                                                                            
/usr/lib/python3.4/site-packages/astropy/coordinates/sky_coordinate.py in __init__(self, *args, **kwargs)                                                                                                                                   
    148                                                                                                                                                                                                                                     
    149         # Finally make the internal coordinate object.                                                                                                                                                                              
--> 150         self._sky_coord_frame = frame.__class__(**coord_kwargs)                                                                                                                                                                     
    151                                                                                                                                                                                                                                     
    152         if not self._sky_coord_frame.has_data:                                                                                                                                                                                      
                                                                                                                                                                                                                                            
/usr/lib/python3.4/site-packages/astropy/coordinates/baseframe.py in __init__(self, *args, **kwargs)                                                                                                                                        
    445                 if (self.representation == SphericalRepresentation and                                                                                                                                                              
    446                         'distance' not in repr_kwargs):                                                                                                                                                                             
--> 447                     representation_data = UnitSphericalRepresentation(**repr_kwargs)                                                                                                                                                
    448                 else:                                                                                                                                                                                                               
    449                     representation_data = self.representation(**repr_kwargs)                                                                                                                                                        
                                                                                                                                                                                                                                            
/usr/lib/python3.4/site-packages/astropy/coordinates/representation.py in __init__(self, lon, lat, copy)                                                                                                                                    
    430             lon, lat = broadcast_quantity(lon, lat)                                                                                                                                                                                 
    431         except ValueError:                                                                                                                                                                                                          
--> 432             raise ValueError("Input parameters lon and lat cannot be broadcast")                                                                                                                                                    
    433                                                                                                                                                                                                                                     
    434         self._lon = lon                                                                                                                                                                                                             
                                                                                                                                                                                                                                            
ValueError: Input parameters lon and lat cannot be broadcast

**Comment by embray** [comment]
(Note, I copied the traceback into the issue description so it's all in one place)

**Comment by embray** [comment]
@mboquien If that's the case, that should be reported as an issue against astroquery.  It should not be returning `bytes` objects.  The issue for Astropy in that case, then, is that the error message is misleading.

**Comment by keflavich** [comment]
@embray - is there any reason `SkyCoord` doesn't accept byte strings?  This may be an easy fix in astroquery but I'm surprised `SkyCoord` is so inflexible.

**Comment by keflavich** [comment]
Following up, I think at least naively astroquery *should* be returning bytes, since the data returned from SIMBAD is ASCII.  The `bytes` return type is actually occurring because astropy's `votable.parse` is reading a `BytesIO` object, and I'm not sure it makes sense (or is even possible) to change that.  I admit, though, whenever we get into this unicode vs. byte issue, I get completely lost about which tools support which data types.

**Comment by embray** [comment]
In general user-facing APIs should never deal with bytes objects unless those APIs are specifically related to binary data.

**Comment by embray** [comment]
To put it another way, the fact that you can write a `bytes` object literal like `b'15 09 14.3'` exists mostly as a convenience for developers of ASCII-based protocols.  But it does not mean that the resulting object should be considered in any way equivalent to the string `'15 09 14.3'`, hence seeming oddities like

```
>>> b'15 09 14.3'[0]
49
```

The place where this gets tricky and annoying for our purposes is how unicode is currently handled in Numpy (read: not well).

**Comment by taldcroft** [comment]
I agree with @embray that this is fundamentally an upstream issue in astroquery.  As soon as you starting putting out `bytes` objects to users then things will start breaking (for instance `'15' in b'15 20'` fails, in addition to Erik's example).  

We should definitely try to make exceptions more informative, but my feeling is that we don't want to go down the slippery slope of patching astropy to accept `bytes` objects instead of `str` in Python 3.

**Comment by taldcroft** [comment]
@keflavich - you might be interested in [convert_bytestring_to_unicode]( http://astropy.readthedocs.org/en/latest/api/astropy.table.Table.html#astropy.table.Table.convert_bytestring_to_unicode)

**Comment by keflavich** [comment]
@taldcroft The problem actually comes from `votable.parse`: it won't accept a `StringIO` object; it insists on having bytes.  I haven't managed to track down the error explicitly, but the exception I get if I try passing a string (rather than byte) object is `*** TypeError: expected bytes, str found`.  So, is there a way to get VOTables to return stringy things instead of bytey things?

**Comment by astrofrog** [comment]
@keflavich - I think what @taldcroft means is that if you do ``t = Table.read('blah.xml', format='votable')`` then ``t.convert_bytestring_to_unicode()`` then it should give you what you need?

**Comment by astrofrog** [comment]
cc @mdboom for the VO table issue - seems like it should be able to accept str, not just bytes

**Comment by taldcroft** [comment]
For the record, this is a demonstration of the VO table issue in Python 3.4:
```
In [29]: text = open('astropy/io/votable/tests/data/irsa-nph-m31.xml').read()
In [30]: type(text)
Out[30]: str

In [31]: fh = io.StringIO(text)
In [32]: type(fh.getvalue())
Out[32]: str

In [33]: vot = parse(fh)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-33-d876f09a0a39> in <module>()
----> 1 vot = parse(fh)

/Users/aldcroft/anaconda/envs/py34-astropy1.0/lib/python3.4/site-packages/astropy/io/votable/table.py in parse(source, columns, invalid, pedantic, chunk_size, table_number, table_id, filename, unit_format, datatype_mapping, _debug_python_based_parser)
    144         _debug_python_based_parser=_debug_python_based_parser) as iterator:
    145         return tree.VOTableFile(
--> 146             config=config, pos=(1, 1)).parse(iterator, config)
    147 
    148 

/Users/aldcroft/anaconda/envs/py34-astropy1.0/lib/python3.4/site-packages/astropy/io/votable/tree.py in parse(self, iterator, config)
   3297         config['_current_table_number'] = 0
   3298 
-> 3299         for start, tag, data, pos in iterator:
   3300             if start:
   3301                 if tag == 'xml':

TypeError: expected bytes, str found

```

**Comment by keflavich** [comment]
Thanks @astrofrog, @taldcroft:  I had simply not looked at the code closely enough to realized that it does exactly the right thing.  This is an easy fix now and I've implemented it (though it took an hour instead of 1 minute because of hidden away error messages... my own fault).

**Comment by eteq** [comment]
@keflavich @EricDepagne - it seems to me like this was fixed in astropy/astroquery#481, so I'm going to close this.  But if I mis-understood and some part of this is still valid, feel free to re-open.
